ddtrace/tracer/textmap.go

```diff
@@ -948,6 +954,13 @@ func parseTracestate(ctx *spanContext, header string) {
 				}
 				setPropagatingTag(ctx, keyDecisionMaker, val)
 			} else if strings.HasPrefix(key, "t.") {
+				if key == "t.tid" {
+					if ok := validIDRgx.MatchString(val); !ok {
+						ctx.trace.setTag(keyPropagationError, fmt.Sprintf("malformed_tid %s", val))
+					} else if val != ctx.traceID.UpperHex() {
+						ctx.trace.setTag(keyPropagationError, fmt.Sprintf("inconsistent_tid %s", val))
+					}
+				}
```

Comment:
(From Slack convo) this is all optional, right? Based on the RFC and comments, it sounds like it's perfectly valid to just not propagate/use the `_dd.p.tid` tag when using W3C trace context propagation. If this is optional, then let's not do this unless we have a very compelling reason that it'll help customers. The code is already fairly complicated, so let's avoid doing more than we need to.

i.e. you can just add a continue right here instead if `key == "t.tid"` to not propagate it.

Reply:
I'm not sure I see what you mean. Is that what you suggest ? : 
```go
   if key == "t.tid" && (val != ctx.traceID.UpperHex() || (!validIDRgx.MatchString(val))) {
	continue
    }
     setPropagatingTag(key, val)
```